tests/test_objects.py

```diff
@@ -57,6 +59,36 @@ def test_objects():
        #pickles(member, exact=True)
         pickles(member, exact=False)
 
+test_pycapsule = None
+
+if pickle._dill.HAS_CTYPES:
+    import ctypes
+    if hasattr(ctypes, 'pythonapi'):
+        def test_pycapsule():
+            name = ctypes.create_string_buffer(b'dill._testcapsule')
+            capsule = pickle._dill._PyCapsule_New(
+                ctypes.cast(pickle._dill._PyCapsule_New, ctypes.c_void_p),
+                name,
+                None
+            )
+            with warnings.catch_warnings():
+                warnings.simplefilter("ignore")
+                pickle.copy(capsule)
+            pickle._testcapsule = capsule
+            with warnings.catch_warnings():
+                warnings.simplefilter("ignore")
+                pickle.copy(capsule)
+            pickle._testcapsule = None
+            try:
+                with warnings.catch_warnings():
+                    warnings.simplefilter("ignore", pickle.PicklingWarning)
+                    pickle.copy(capsule)
+            except pickle.UnpicklingError:
+                pass
+            else:
+                raise AssertionError("Expected a different error")
 
 if __name__ == '__main__':
     test_objects()
+    if test_pycapsule is not None:
```

Comment:
this test should be moved elsewhere. Its own test module is fine.  Note that `PyCapsule` objects are not yet in `dill._objects`.

Reply:
Thanks for the reminder.